### PR TITLE
[fix] don't error when printing nan

### PR DESCRIPTION
**Pull request by kchanqvq**

### CODE REVIEW

**Comment by digikar99** [comment]
How about instead of this, we surround the part that prints with `(float-features:with-float-traps-masked t ...)`? I'm expecting that will also take care of infinities, but not sure if `T` would hide too many errors.

**Comment by kchanqvq** [comment]
That's a good idea! Although, it mess up the tabulation (which infinity currently also does). Do we want to do something to preserve tabulation?

**Comment by kchanqvq** [comment]
How about this solution? This checks for NaN and Inf explicitly, so the nice tabulation is preserved.

**Comment by digikar99** [comment]
I think this looks good. Thanks!

I'm away from keyboard for another few days, but I hope to merge around the weekend.

**Comment by digikar99** [comment]
It seems the signs for `Inf` are opposite of expected. I'm also leaning towards keeping the format directives simple.

```lisp
DENSE-ARRAYS-PLUS-LITE> (let ((a (rand 2 3 :type 'double-float :min -1e9 :max 1e9)))
                          (float-features:with-float-traps-masked t
                            (setf (aref a 0 1) float-features:double-float-positive-infinity)
                            a))
#<STANDARD-DENSE-ARRAY :ROW-MAJOR 2x3 DOUBLE-FLOAT
  ( -7.287d+08    -Inf      -6.966d+08)
  (  6.168d+08  -7.842d+08   4.110d+08)
   {10096218A3}>
DENSE-ARRAYS-PLUS-LITE> (let ((a (rand 2 3 :type 'double-float :min -1e9 :max 1e9)))
                          (float-features:with-float-traps-masked t
                            (setf (aref a 0 1) float-features:double-float-negative-infinity)
                            a))
#<STANDARD-DENSE-ARRAY :ROW-MAJOR 2x3 DOUBLE-FLOAT
  ( -8.768d+08    +Inf       1.167d+08)
  (  4.254d+08   7.360d+08   8.599d+08)
   {1009625FF3}>
```

File: src/dense-arrays.lisp [hunk 1]
```diff
@@ -351,21 +351,24 @@ without copying."
...
-                             "~7,3,,,f    "
-                             "~7,3,,,f    "))
-                  (t      "~s"))
-              arg))))
+    (typecase number
+      (float
+       (cond ((float-features:float-nan-p number)
+              (format stream "    NaN    "))
+             ((float-features:float-infinity-p number)
+              (format stream "   ~:[+~;-~]Inf    " (plusp number)))
```
Comment (digikar99) [inline src/dense-arrays.lisp]:
```suggestion
              (format stream "   ~AInf    " (if (plusp number) #\+ #\-)))
```

**Pull request merged**